src/ews_serialize.erl

```diff
@@ -339,9 +442,30 @@ match_children_elems([], [#elem{meta=#meta{min=0}}=E|Es], [], Res) ->
     match_children_elems([], Es, [], [{undefined,E}|Res]);
 match_children_elems([], [#elem{meta=#meta{min=0}}=E|Es], Acc, Res) ->
     match_children_elems([], Es, [], [{undefined,E},lists:reverse(Acc)|Res]);
+match_children_elems([], [#elem{qname=Name,meta=#meta{min=_N}}|_], _, _) ->
+    error({missing_non_optional_element, Name});
 match_children_elems([], [], Acc, Res) ->
     [lists:reverse(Acc) | lists:reverse(Res)].
 
+validate_attrs([{?SCHEMA_INSTANCE_NS, _} | As], PossAttrs, Acc) ->
+    validate_attrs(As, PossAttrs, Acc);
+validate_attrs([{Name, Value} | As], PossAttrs, Acc) ->
+    case [ P || #attribute{name = {_, N}} = P <- PossAttrs, N == Name ] of
+        [] ->
+            logger:notice("Unexpected attribute: ~p~n", [Name]),
+            validate_attrs(As, PossAttrs, Acc);
+        [#attribute{}] ->
+            %% TODO: validate the type
+            %% TODO: how to we handle utf8 in both Name and Value?
+            %% By compiling with debug_info the typespec in the records should
+            %% load all possible atoms.
+            NameAtom = list_to_existing_atom(Name),
```

Comment:
So, we have to use `list_to_existing_atom/1` since we are taking in stuff from the internets.

And it turned out that the typespec that is emitted is enough to get all the atoms into the running erl system by just loading a module that has the emitted header file included.

We don't actually know if it will break if you strip the module? it might be that all the typespec atoms stays in a stripped module, but we have a hunch it might, but again we are not sure.

So in the end I decided not to add the dummy function with all the atoms to the hrl files, since it might possibly break only if you strip your modules, and your code doesn't try to parse any of the attributes that are being sent, so a big if.

This has been tested by starting a new erl and load an already compiled module that includes the hrl file and all atoms were loaded.